### DISCUSSION TITLE
Episode 184: Safety in Swift 6, Protocols & More with Doug Gregor

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#79] [Episode]
Episode Link: https://adspthepodcast.com/2024/05/31/Episode-184.html

Feel free to write comments or ask questions 🙂

**Comment by Brian-M-J** [comment]
I found a Doug Gregor gem: this [talk](https://www.microsoft.com/en-us/research/video/generic-programming-in-the-parallel-boost-graph-library/) on "Generic Programming in the Parallel Boost Graph Library" given all the way back in 2007.

**Comment by ole** [comment]
I really enjoyed these episodes, thank you to the guest and the hosts!

The PLDI 2022 paper @DougGregor credits (at ~8:25) as the basis for Swift’s concurrency solution for reference types is [A Flexible Type System for Fearless Concurrency](https://www.cs.cornell.edu/andru/papers/gallifrey-types/) by Mae Milano, Joshua Turcotti, and Andrew C. Myers. (It's also referenced in [Swift Evolution SE-0414: Region-based Isolation](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0414-region-based-isolation.md).)

@codereport Maybe you could add this link to the shownotes?

**Reply by codereport** [reply]
Done! I linked the wrong paper that also had "Fearless Concurrency" in the title! Thanks!

**Reply by ole** [reply]
Great, thanks!
